### ISSUE TITLE
[FEATURE REQUEST] Add Processing Details to Output Video Filename and/or Metadata

### ISSUE THREAD

**Issue by GautDlpr** [k4yt3x/video2x#1264]
When generating the output video, it would be beneficial to include processing details such as mode, filters, scaling factor, and the model used.
These details could replace the default suffix .processed in the output filename and/or be embedded in the metadata of the video file for reference.

Benefits:
- Improved traceability: Users can quickly identify the applied settings by looking at the filename or metadata.
- Better organization: Avoid confusion when handling multiple processed videos.
- Ease of debugging: The embedded metadata makes it easier to verify processing parameters post-export.

This feature could be optional and controlled via a setting in desktop version and command-line flag in CLI for users who prefer the simpler naming scheme.

**Comment by k4yt3x** [comment]
Yeah I've thought about this feature. I'm definitely going to add something that will allow you to control the output filename. It's just one of the nice-to-have things that'll be added progressively into the software.

**Comment by k4yt3x** [comment]
In 6.3.0 I've already replaced `.processed` with the name of the processor. If necessary, we can implement a custom formatter for each processor that makes a suffix, like `output.realesrgan_4x_realesr-animevideov3.mkv`. Although I'm wondering if this will be too long. Perhaps we give the user a toggle and provide a field to enter custom prefix/suffixes.

**Comment by k4yt3x** [comment]
In addition to implementing this in the GUI, I also want to make the output file path an optional parameter and be able to generate a file name when it's not set.
